### DISCUSSION TITLE
Waveform and Timeline plugin not scrolling in sync.

### DISCUSSION THREAD

**Discussion by ruchaheda** [katspaugh/wavesurfer.js#3901] [General]
I am attempting to initialize a waveform and timeline in the same container. When I set the width using viewport width (to avoid horizontal overflow), the timeline stops syncing with the waveform.

Furthermore, the timeline disappears as I scroll right within the viewport width.

The closest issue I found in bugs to this was: https://github.com/katspaugh/wavesurfer.js/issues/3782. However, there doesn't seem to be a fix. 

Is anyone else experiencing this and have you found a solution that works with React/Typescript?

Here is my waveform, timeline, and component implementation - 

Timeline definition:
```
const topTimeline = TimelinePlugin.create({
            container: '#waveform',
            height: 20,
            timeInterval: 1,
            // insertPosition: 'beforebegin',
            primaryLabelInterval: 5,
            secondaryLabelOpacity: 0,
            style: {
                fontSize: '20px',
                color: '#2D5B88',
            },
        })
```

Waveform initialization:
```
const wavesurfer = WaveSurfer.create({
            container: '#waveform',
            waveColor: 'rgb(200, 0, 200)',
            progressColor: 'rgb(100, 0, 100)',
            autoScroll: true,
            autoCenter: true,    // Disable auto-centering
            minPxPerSec: 20,      // Minimum pixels per second for the waveform
            fillParent: false,
            height: 100,
            barWidth: 2,
            barGap: 1,
            barRadius: 2,
            plugins: [topTimeline, hover, regionPlugin]
        });
```

Component implementation:
```
<Box
                sx={{ 
                    width: '100vw',
                    overflow: 'hidden',   // Prevent the outer box from scrolling
                    position: 'relative', // Stable layout for the waveform
                }}
            >
                <Box 
                    id="waveform" 
                    sx={{
                        width: '100%',
                        height: '150px',
                        overflowX: 'auto',     // Enable horizontal scrolling within this container
                        whiteSpace: 'nowrap',  // Prevent wrapping to enable horizontal scrolling
                      }}
                />
            </Box>
```

Thanks in advance for your help!

**Comment by katspaugh** [comment]
You shouldn’t specify a container for the timeline. Just let it render inside the default container (which is the scrolling element inside the wavesurfer div).

**Comment by ruchaheda** [comment]
Thanks. I switched back to using a div and removed the 'container' attribute in the timeline's instantiation and I think this fixed the issue. I appreciate the help!
